rest/route/project.go

```diff
@@ -185,3 +186,133 @@ func (h *versionsGetHandler) Run(ctx context.Context) gimlet.Responder {
 
 	return gimlet.NewJSONResponse(versions)
 }
+
+/*
+ * Creates project (project_ref)
+ * PUT: /rest/v2/projects/
+ * Perm: superUser
+ */
+type projectCreateHandler struct {
+	projectRef *model.APIProjectRef
+
+	sc data.Connector
+}
+
+func makeProjectCreateRoute(sc data.Connector) gimlet.RouteHandler {
+	return &projectCreateHandler{sc: sc}
+}
+
+func (h *projectCreateHandler) Factory() gimlet.RouteHandler {
+	return &projectCreateHandler{sc: h.sc}
+}
+
+func (h *projectCreateHandler) Parse(ctx context.Context, r *http.Request) error {
+	body := util.NewRequestReader(r)
+	defer body.Close()
+
+	if err := util.ReadJSONInto(body, &h.projectRef); err != nil {
+		return errors.Wrap(err, "Cannot parse the content as JSON!")
+	}
+
+	return nil
+}
+
+func (h *projectCreateHandler) Run(ctx context.Context) gimlet.Responder {
+	createdProject, err := h.sc.CreateProject(h.projectRef)
+
+	if err != nil {
+		return gimlet.MakeJSONErrorResponder(errors.Wrap(err, "Cannot create project!"))
+	}
+
+	apiProject := &model.APIProject{}
+	// The method doesn't accept pointers, we have to dereference it first
+	err = apiProject.BuildFromService(*createdProject)
+
+	if err != nil {
+		return gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
+			Message:    "problem converting project document",
+			StatusCode: http.StatusInternalServerError,
+		})
+	}
+
+	resp := gimlet.NewJSONResponse(apiProject)
+	err = resp.SetStatus(http.StatusCreated)
+
+	if err != nil {
+		return gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
+			Message:    "Cannot set status code",
+			StatusCode: http.StatusInternalServerError,
+		})
+	}
+
+	return resp
+}
+
+/*
+ * Updates project (project_ref) with given identifier
+ * PATCH: /rest/v2/projects/{project_id}
+ * Perm: superUser
+ */
+type projectUpdateHandler struct {
+	projectRef *model.APIProjectRef
+
+	sc data.Connector
+}
+
+func makeProjectUpdateRoute(sc data.Connector) gimlet.RouteHandler {
+	return &projectUpdateHandler{sc: sc}
+}
+
+func (h *projectUpdateHandler) Factory() gimlet.RouteHandler {
+	return &projectUpdateHandler{sc: h.sc}
+}
+
+func (h *projectUpdateHandler) Parse(ctx context.Context, r *http.Request) error {
+	// Read raw bytes from request body
+	body := util.NewRequestReader(r)
+	defer body.Close()
+
+	projectRef := MustHaveProjectContext(ctx).ProjectRef
+	if projectRef == nil {
+		return errors.New("Cannot process request")
+	}
+
+	projectId := projectRef.Identifier
+
+	// Initialize the API model with an empty entity
+	h.projectRef = &model.APIProjectRef{}
+
+	if err := h.projectRef.BuildFromService(*projectRef); err != nil {
+		return errors.Wrap(err, "Cannot process request")
+	}
+
+	if err := util.ReadJSONInto(body, &h.projectRef); err != nil {
+		return errors.Wrap(err, "JSON format or content is invalid!")
+	}
+
+	*h.projectRef.Identifier = projectId
```

Comment:
I think that if you change BuildFromService and remove line 257, you will not need to dereference these pointers.